tests/std/tests/GH_000685_condition_variable_any/test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Test GH#685 "wait_for in condition_variable_any should unlock and lock"
+
+#include <cassert>
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+
+using namespace std;
+
+namespace { // anonymous namespace
+    class my_mutex { // user-defined mutex type
+    public:
+        void lock() {
+            mtx.lock();
+            ++num_lock;
+        }
+
+        void unlock() {
+            mtx.unlock();
+        }
+
+        inline int num_locks() {

```

Comment:
Member functions defined within their class definition are already implicitly `inline`. Also, this should probably be `const` as it's a pure observer:

```suggestion
        int num_locks() const {
```